libr/debug/p/native/bsd/bsd_debug.c

```diff
@@ -498,3 +498,39 @@ RList *bsd_desc_list(int pid) {
 	return ret;
 #endif
 }
+
+RList *bsd_thread_list(RDebug *dbg, int pid, RList *list) {
+#if __KFBSD__
+	int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PID | KERN_PROC_INC_THREAD, pid };
+	struct kinfo_proc *kp;
+	size_t len = 0;
+	size_t max;
+	int i = 0;
+
+	if (sysctl (mib, 4, NULL, &len, NULL, 0) == -1) {
+		r_list_free (list);
+		return NULL;
+	}
+
+	len += sizeof(*kp) + len / 10;
+	kp = malloc(len);
+	if (sysctl (mib, 4, kp, &len, NULL, 0) == -1) {
+		free (kp);
+		r_list_free (list);
+		return NULL;
+	}
+
+	max = len / sizeof(*kp);
+	for (i = 0; i < max; i ++) {
+		RDebugPid *pid_info;
+		pid_info = r_debug_pid_new (NULL, kp[i].ki_tid, kp[i].ki_uid, 's', 0);
+		r_list_append (list, pid_info);
+	}
+
+	free (kp);
+	return list;
+#else
+	r_list_free (list);
```

Comment:
Maybe add a warning saying "bsd_thread_list: Unsupported architecture" for anything that isn't KFBSD?

Reply:
unsupported platform most likely I might say but yes.

Reply:
Yeah wrong phrasing on my part